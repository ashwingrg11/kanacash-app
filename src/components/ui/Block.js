import React from 'react';
import {View, StyleSheet, TouchableOpacity} from 'react-native';
import PropTypes from 'prop-types';
import theme from '~/components/theme/Style';
import {MediumText, RegularText, SemiBoldText} from '~/components/ui/Text';

export default function Block({
  left,
  primary,
  title,
  caption,
  leftContent,
  rightContent,
  onPress,
  backgroundColor,
  rightContentStyle,
}) {
  const invert = primary ? primary : backgroundColor;
  return (
    <TouchableOpacity onPress={onPress}>
      <View style={[styles.container(primary, backgroundColor)]}>
        {leftContent}
        <View style={styles.textWrapper}>
          <View style={{flexDirection: 'row'}}>
            <View style={{}}>
              {caption ? (
                <MediumText text={title} invert={invert} />
              ) : (
                <SemiBoldText text={title} invert={invert} />
              )}
              {caption && <RegularText text={caption} invert={invert} />}
            </View>
            <View style={[styles.rightContentStyle, rightContentStyle]}>
              {rightContent}
            </View>
          </View>
        </View>
      </View>
    </TouchableOpacity>
  );
}

Block.defaultProps = {
  primary: false,
};

Block.propTypes = {
  leftContent: PropTypes.element,
  rightContent: PropTypes.element,
  onPress: PropTypes.func,
};

const styles = StyleSheet.create({
  container: (primary, backgroundColor) => ({
    borderRadius: theme.defaultRadius,
    backgroundColor: primary
      ? theme.primaryColor
      : backgroundColor
      ? backgroundColor
      : theme.white,
    paddingHorizontal: 10,
    paddingVertical: 12,
    flexDirection: 'row',
    alignItems: 'center',
    marginVertical: 5,
    justifyContent: 'space-between',
  }),
  textWrapper: {
    marginHorizontal: 10,
    flex: 3,
  },
  rightContentStyle: {
    flex: 1,
    alignItems: 'flex-end',
  },
  statusStyle: {
    flexDirection: 'row',
    alignItems: 'center',
    marginBottom: 5,
  },
});
