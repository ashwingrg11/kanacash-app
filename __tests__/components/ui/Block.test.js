import 'react-native';
import React from 'react';

import Block from '~/components/ui/Block';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

describe('<Block/>', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Block backgroundColor="red" title="title"
    caption="caption"/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
