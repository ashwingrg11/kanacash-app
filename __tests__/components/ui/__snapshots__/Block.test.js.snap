// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Block/> renders correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "red",
          "borderRadius": 5,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginVertical": 5,
          "paddingHorizontal": 10,
          "paddingVertical": 12,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "flex": 3,
          "marginHorizontal": 10,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <View
          style={Object {}}
        >
          <Text
            invert="red"
            style={
              Array [
                Object {
                  "color": "#1F1F1F",
                  "fontFamily": "OpenSans-Bold",
                  "fontSize": 18,
                  "fontWeight": "500",
                  "lineHeight": 22,
                },
                Object {
                  "color": "#fff",
                },
                undefined,
              ]
            }
            text="title"
          >
            title
          </Text>
          <Text
            invert="red"
            style={
              Array [
                Object {
                  "color": "#5C5C5C",
                  "fontFamily": "OpenSans-Regular",
                  "fontSize": 16,
                  "lineHeight": 20,
                },
                Object {
                  "color": "#fff",
                },
                undefined,
              ]
            }
            text="caption"
          >
            caption
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "flex-end",
                "flex": 1,
              },
              undefined,
            ]
          }
        />
      </View>
    </View>
  </View>
</View>
`;
